df.world-data.xml

```diff
@@ -703,18 +703,20 @@
         <int32_t name="year_tick"/>
 
         -- v0.40.14:
-        <stl-vector name='unk_c0' type-name='int16_t' comment="6 bytes allocated seen. Cannot be pointer"/>
-        <stl-vector name='unk_d0' type-name='int16_t' comment="6 bytes allocated seen. Cannot be pointer"/>
-        <stl-vector name='unk_e0' type-name='int16_t' comment="6 bytes allocated seen. Cannot be pointer. Seen on the same element, so c0, d0, and e0 may be connected"/>
-        <stl-vector name='unk_f0' type-name='int32_t' comment="4 bytes allocated seen. Cannot be pointer"/>
-        <stl-vector name='unk_100' type-name='int32_t' comment="4 bytes allocated seen. Cannot be pointer"/>
-        <stl-vector name='unk_110' type-name='int32_t' comment="4 bytes allocated seen. Cannot be pointer. Seen on the same element, so f0, 100, and 110 may be connected"/>
+        <compound name='picked_growths' comment="also includes 'automatically picked' i.e. fallen fruit that becomes item_spatter">
+            <stl-vector name='x' type-name='int16_t' comment="within the MLT"/>
+            <stl-vector name='y' type-name='int16_t' comment="within the MLT"/>
+            <stl-vector name='z' type-name='int16_t' comment="uncertain coordinate system. 100 = surface on my flat embark, 101 in the air, while the normal coordinates don't go that high"/>
+            <stl-vector name='subtype' type-name='int32_t' comment="subtype of the growth within the raws of the implicit plant"/>
+            <stl-vector name='density' type-name='int32_t' comment="unknown why the raw density field is placed here, as there's nothing left"/>
```

Comment:
Perhaps this allows for _partial_ harvests of extremely dense growths, so you can pick the rest of it later.

Reply:
Possibly, although I've seen only 1, possibly 2, and 1000 here. If your idea is correct we'd briefly see something in between 2 and 1000 for trees, but that would require a bit of luck (or care) with the timing.
I haven't seen anything indicating a gradual picking graphically, but the "graphics" might not show the full picture, and I haven't studied fruit picking for hours...

Reply:
It's possible that it only happens when gathering items in Adventurer mode - I'd try it myself, but I don't have a script handy for finding the correct object_data entry for my current location.

Reply:
´´´
  local embark_x = df.global.world.world_data.region_details [0].pos.x
  local embark_y = df.global.world.world_data.region_details [0].pos.y
  local embark_min_x, embark_max_x, embark_min_y, embark_max_y
  local object_id
  
  for i = 0, 15 do
    for k = 0, 15 do
      if df.global.world.world_data.region_details [0].biome [i] [k] >= 48 then
        if not embark_min_x then
          embark_min_x = i
        end
        
        embark_max_x = i
        
        if not embark_min_y then
          embark_min_y = k
        end
        
        embark_max_y = k

        object_id = i + embark_x * 16 + k * 16 * df.global.world.world_data.world_width + embark_y * 256 * df.global.world.world_data.world_width
        dfhack.println (df.world_object_data.find (object_id), object_id)        
      end
    end
  end
´´´
This snippet gets the object ids for an embark. It obviously would have to be adapted to adventurer mode, where I guess the region_details [0] part remains the same, but there'd have to be some other logic to find the current MLT. I'm rather unfamiliar with adventurer mode, though.

Reply:
I tried a more brute-force approach and simply searched through _all_ world_object_data records to find ones with non-empty picked_growths lists, and it looks like Adventurer mode just doesn't use them at all - no matter how many finger limes I picked from a nearby tree, all of those lists remained empty.

Reply:
Brute force is what I've used mostly, until trying a callback every frame, where the frame rate dropped to 28 even when accessing only the appropriate entries.  It's odd, however, that different structures would be used for the same purpose in Adventurer mode and Fortress mode, as I assume the trees actually were picked visually (and not possible to pick indefinitely).

A fruit tree density of 1000 looks like some kind of max value rather than a sensible one, when shrubs tend to have a value of 1, so it might be that partial picking was waiting for the agriculture overhaul.

Have you used tools to look at what code Adventurer mode picking uses? It might be something different that uses different data structures.